### PR TITLE
Replace AnalyticsTracker with AnalyticsTrackerWrapper, write tests in OrderFullFillViewModel

### LINKED ISSUE #6752: Replace hardcoded AnalyticsTracker with AnalyticsTrackerWrapper (by injecting it) in OrderFullFillViewModel
Replace hardcoded AnalyticsTracker with AnalyticsTrackerWrapper (by injecting it) in OrderFullFillViewModel. Doing this change helps us in writing unit tests that verify proper tracks events are being fired as expected with appropriate parameters.

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #6752 

### Description

Replace hardcoded AnalyticsTracker with AnalyticsTrackerWrapper (by injecting it) in OrderFullFillViewModel. Doing this change helps us in writing unit tests that verify proper tracks events are being fired as expected with appropriate parameters.

Additionally, this PR also adds tests that verify that proper UI events along with the proper message are being triggered when the shipment tracking is deleted successfully/failure

### Testing instructions

Ensure that the order fulfillment screen and its logic behave as expected.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/fulfill/OrderFulfillViewModel.kt [hunk 1]
```diff
@@ -192,17 +195,18 @@ class OrderFulfillViewModel @Inject constructor(
         _shipmentTrackings.value = shipmentTrackings
     }
 
-    private fun deleteOrderShipmentTracking(shipmentTracking: OrderShipmentTracking) {
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/fulfill/OrderFulfillViewModel.kt:198]: 💡 Please let me know what you think about this change. If you think making a method public for the sake of testing is not worth it. We can undo this change.

The Android studio warns the developer when this method is being used anywhere else apart from this file though.

**Review by rossanafmenezes — APPROVED** [review]
Looks good to me
